### ISSUE TITLE
0.18.0 some modules are missing

### ISSUE THREAD

**Issue by midfinup1** [pylakey/aiotdlib#17]
Traceback (most recent call last):
  File "\Telegram_Bot\Bot.py", line 3, in <module>
    from aiotdlib.api.types import ReplyMarkupInlineKeyboard, InlineKeyboardButton
  File "\lib\site-packages\aiotdlib\__init__.py", line 3, in <module>
    from .client import (
  File "\lib\site-packages\aiotdlib\client.py", line 31, in <module>
    from .api import (
  File "\lib\site-packages\aiotdlib\api\__init__.py", line 7, in <module>
    from .api import API
  File "\venv\lib\site-packages\aiotdlib\api\api.py", line 10, in <module>
    from .functions import *
  File "\lib\site-packages\aiotdlib\api\functions\__init__.py", line 7, in <module>
    from .accept_call import AcceptCall
  File "\lib\site-packages\aiotdlib\api\functions\accept_call.py", line 11, in <module>
    from ..types import CallProtocol
  File "\lib\site-packages\aiotdlib\api\types\__init__.py", line 15, in <module>
    from .attachment_menu_bot import AttachmentMenuBot
ModuleNotFoundError: No module named 'aiotdlib.api.types.attachment_menu_bot'

How can I fix this?

**Comment by pylakey** [comment]
Could you provide a minimal code example?

**Comment by midfinup1** [comment]
Sure

`client_path = 'C:\\Users\\midfinup\\PycharmProjects\\SpotiFind\\td\\tdlib\\bin\\tdjson.dll'
bot = Client(bot_token=token, api_id=api_id, api_hash=api_hash, library_path=client_path)

@bot.bot_command_handler(command='start')
def send_welcome(message_):
    message = message_
    keyboard = ReplyMarkupInlineKeyboard()
    keyboard.row_width = 2
    keyboard.add(InlineKeyboardButton(text="Start", callback_data="/start"))
    keyboard.add(InlineKeyboardButton(text="Info", callback_data="/info"),
                 InlineKeyboardButton(text="Details", callback_data="/help"))
    bot.send_sticker(message.chat.id, 'CAACAgIAAxkBAAEBd-9g0KBzF4nMZr8kNTyGnaqW7vKg2gAC8wMAAlrmjhFXAAESHYTiOwkfBA')
    bot.send_message(message.chat.id, f'Ready to go 😊\n\t',
                     reply_markup=keyboard)

bot.run()`

**Comment by MrOzOn** [comment]
Similar problem.
I can't run base example on my MBP 2020 with Intel

ModuleNotFoundError: No module named 'aiotdlib.api.types.attachment_menu_bot'

**Comment by jsonguard** [comment]
Okay, i found the problem root:
Generated python code in pypi package is broken

IDE founds some errors in file:
aiotdlib/aiotdlib/api/functions/__init__.py

I clone repo, stand on tag 0.18.0 and run `poetry run aiotdlib_generator `
It makes too many changes but IDE didnt found problem in generated code
Then I build package with `poetry build` command, install it and now demo example works great!

@pylakey 
Could you rebuild pypi package correctly?
Thank you in advance!

**Comment by ToyPoodleM** [comment]
Hi. jesnguard.
I'm facing the same problem as you, and I think you've found a solution.
But I can't apply the solution you wrote down.
Can you tell me how to do it in detail?

**Comment by jsonguard** [comment]
> Hi. jesnguard. I'm facing the same problem as you, and I think you've found a solution. But I can't apply the solution you wrote down. Can you tell me how to do it in detail?

Hello! 
To work with aiotdlib you need to install poetry
https://python-poetry.org/docs/
Poetry also is available as deb/rpm package in most linux distributions

Now you can build your own package

0. Clone this repo
```bash
git clone https://github.com/pylakey/aiotdlib.git
cd aiotdlib
git checkout tags/0.18.0
```

1. Install dependencies
```bash
poetry install
```

2. Generate code
```bash
poetry run aiotdlib_generator
```

3. Build package
```bash
poetry build
```

Done
You can find fixed packages in `dist/`  directory

**Comment by ToyPoodleM** [comment]
Thank you for your quick response.
However, I am a Windows user.
Do you think the aiotdlib package is suitable for use in Windows?
If the method used in Windows is the same, I will try it. 
In Windows, I tried to install Poetry through Pip and build there, but it didn't work well, but I will try again.

**Comment by ToyPoodleM** [comment]
> > Hi. jesnguard. I'm facing the same problem as you, and I think you've found a solution. But I can't apply the solution you wrote down. Can you tell me how to do it in detail?
> 
> Hello! To work with aiotdlib you need to install poetry https://python-poetry.org/docs/ Poetry also is available as deb/rpm package in most linux distributions
> 
> Now you can build your own package
> 
> 0. Clone this repo
> 
> ```shell
> git clone https://github.com/pylakey/aiotdlib.git
> cd aiotdlib
> git checkout tags/0.18.0
> ```
> 
> 1. Install dependencies
> 
> ```shell
> poetry install
> ```
> 
> 2. Generate code
> 
> ```shell
> poetry run aiotdlib_generator
> ```
> 
> 3. Build package
> 
> ```shell
> poetry build
> ```
> 
> Done You can find fixed packages in `dist/` directory

Thank you for your quick response.
However, I am a Windows user.
Do you think the aiotdlib package is suitable for use in Windows?
If the method used in Windows is the same, I will try it.
In Windows, I tried to install Poetry through Pip and build there, but it didn't work well, but I will try again.

**Comment by ToyPoodleM** [comment]
> ```shell
> poetry build
> ```
SUPER Thanks to 
I solved problem.

**Comment by ToyPoodleM** [comment]
> > > > Hi. jesnguard. I'm facing the same problem as you, and I think you've found a solution. But I can't apply the solution you wrote down. Can you tell me how to do it in detail?
> > > 
> > > 
> > > Hello! To work with aiotdlib you need to install poetry https://python-poetry.org/docs/ Poetry also is available as deb/rpm package in most linux distributions
> > > Now you can build your own package
> > > 
> > > 0. Clone this repo
> > > 
> > > ```shell
> > > git clone https://github.com/pylakey/aiotdlib.git
> > > cd aiotdlib
> > > git checkout tags/0.18.0
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > 1. Install dependencies
> > > 
> > > ```shell
> > > poetry install
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > 2. Generate code
> > > 
> > > ```shell
> > > poetry run aiotdlib_generator
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > 3. Build package
> > > 
> > > ```shell
> > > poetry build
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Done You can find fixed packages in `dist/` directory
> > 
> > 
> > Thank you for your quick response. However, I am a Windows user. Do you think the aiotdlib package is suitable for use in Windows? If the method used in Windows is the same, I will try it. In Windows, I tried to install Poetry through Pip and build there, but it didn't work well, but I will try again.
> 
> To be honest I have never tried doing this on windows, but most likely the package will be built However, the problem is that the package does not contain binary dll of tdlib for windows, so you need to build it manually by commit [pylakey/td@f295ef3](https://github.com/pylakey/td/commit/f295ef3a0d3545970bfd658c3443496be3d28397) Build instruction is available here: https://tdlib.github.io/td/build.html
> 
> Also, you can try Windows Subsystem for Linux (WSL) if you know how it works

Can I ask you one more question?
        api_id=API_ID,
        api_hash=API_HASH,
        phone_number=PHONE_NUMBER
Is there any other way than this way?

I want to use only api_id and hash. 
I'd appreciate it if you could let me know if you knew.

**Comment by pylakey** [comment]
> @pylakey 
> 
> Could you rebuild pypi package correctly?
> 
> Thank you in advance!

Yes, sure, will be fixed in 0.18.1

**Comment by pylakey** [comment]
> Can I ask you one more question?
> 
>         api_id=API_ID,
> 
>         api_hash=API_HASH,
> 
>         phone_number=PHONE_NUMBER
> 
> Is there any other way than this way?
> 
> 
> 
> I want to use only api_id and hash. 
> 
> I'd appreciate it if you could let me know if you knew.

No, you must be authorized using either phone_number (for users accounts) or bot_token (for bots accounts)

**Comment by swimmwatch** [comment]
@pylakey, hello! When will release 0.18.1 version with the fix?

**Comment by pylakey** [comment]
In a few days 0.19.0 will be released with up to date tdlib

**Comment by pylakey** [comment]
Fixed in 0.19.0
